libs/domains/service-helm/feature/src/lib/values-override-yaml-modal/values-override-yaml-modal.tsx

```diff
@@ -0,0 +1,118 @@
+import { type HelmRequestAllOfSource } from 'qovery-typescript-axios'
+import { useEffect } from 'react'
+import { Controller, FormProvider, useForm } from 'react-hook-form'
+import {
+  BlockContent,
+  CodeEditor,
+  CopyToClipboard,
+  Icon,
+  IconAwesomeEnum,
+  LoaderSpinner,
+  ModalCrud,
+} from '@qovery/shared/ui'
+import useCreateHelmDefaultValues from '../hooks/use-create-helm-default-values/use-create-helm-default-values'
+
+export interface ValuesOverrideYamlModalProps {
+  environmentId: string
+  source: HelmRequestAllOfSource
+  onSubmit: (value?: string) => void
+  onClose: () => void
+  content?: string
+}
+
+export function ValuesOverrideYamlModal({
+  source,
+  environmentId,
+  onClose,
+  onSubmit,
+  content,
+}: ValuesOverrideYamlModalProps) {
+  const {
+    data: helmDefaultValues,
+    mutateAsync: createHelmDefaultValues,
+    isLoading: isLoadingHelmDefaultValues,
+    isError: isErrorHelmDefaultValues,
+  } = useCreateHelmDefaultValues()
+
+  const methods = useForm({
+    mode: 'onChange',
+    defaultValues: {
+      content,
+    },
+  })
+
+  useEffect(() => {
+    async function fetchHelmDefaultValues() {
```

Comment:
Hum, this looks weird.
Let's talk about it to see if there is no better solution.